doc/decisions/backend_orm_crate.md

```diff
@@ -0,0 +1,70 @@
+# Backend ORM/SQL crate
+
+## Problem
+
+We are currently using [diesel](https://github.com/diesel-rs/diesel) as our ORM in Rust.  
+Since diesel doesn't support async out of the box we will look at other alteratives and see if a switch is reasonable.  
+There might also be advantages of other ORM crates that as of writing this have not been considered.
+
+## Constraints
+
+1. Migrations need to be supported
+
+## Assumptions
+
+1. diesel is a solid choice, therefore unless other ORMs have obvious advantages differences will not be looked at in detail
+
+## Solutions
+
+### [diesel](https://github.com/diesel-rs/diesel)
+
+Diesel currently is the most used ORM in the Rust ecosystem.  
+It supports migrations written in SQL that can be executed from command line.
+
+Diesel is the ORM we are using as of writing this decisions.
+
+### [diesel](https://github.com/diesel-rs/diesel) + [diesel_async](https://github.com/weiznich/diesel_async)
+
+diesel_async is a drop-in replacement of corresponding diesel methods providing async support.
+
+### [sea-orm](https://github.com/SeaQL/sea-orm)
+
+Sea-orm is based upon [sqlx](https://github.com/jmoiron/sqlx) and supports async out of the box.  
+It supports [migrations](https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration/) written in Rust.
+
+## Decision
+
+Use diesel + diesel_async. There are no obvious disadvantages to diesel_async and required code changes are minimal.
```

Comment:
Please answer here: What did you check for? Is it as reliable, tested, ...?

Reply:
As far as I can remember it was created by the creator of diesel. I haven't looked at the actual implementation though. According to the examples the only difference to normal diesel is that you import different traits and in the performance metrics there is only a minimal decrease.
I would however say that it cannot be as tested as just diesel itself. It has way less downloads, but like I mentioned I believe it is easier to use async all the way through that having to worry about blocking functions.

Reply:
Please write this in the decision itself. Also if it is feature-complete etc.